src/dimension_slice.c

```diff
@@ -983,6 +1025,216 @@ ts_dimension_slice_insert_multi(DimensionSlice **slices, Size num_slices)
+	orig_mcxt = MemoryContextSwitchTo(work_mcxt);
+
+	for (int constraint_index = 0; constraint_index < constraints->num_constraints;
+		 constraint_index++)
+	{
+		int slice_id;
+		DimensionSlice *slice;
+		ChunkConstraint *constraint = &constraints->constraints[constraint_index];
+		Datum minmax[2];
+		AttrNumber attno;
+		TupleInfo *ti;
+		ScanIterator slice_iterator;
+		const Dimension *dim;
+
+		if (!is_correlated_constraint_dimension(constraint))
+			continue;
+
+		/*
+		 * Find the slice by id. Don't have to lock it because the chunk is
+		 * assumed to be locked when this function is called.
+		 */
+		slice_id = constraint->fd.dimension_slice_id;
+		slice_iterator = ts_dimension_slice_scan_iterator_create(NULL, orig_mcxt);
+		ts_dimension_slice_scan_iterator_set_slice_id(&slice_iterator, slice_id, NULL);
+		ts_scan_iterator_start_scan(&slice_iterator);
+		ti = ts_scan_iterator_next(&slice_iterator);
+		if (ti == NULL)
+			elog(ERROR, "dimension slice %d not found!", slice_id);
+
+		slice = ts_dimension_slice_from_tuple(ti);
+
+		/* get the dimension from the correlated subspace. It has to exist */
+		dim = ts_hyperspace_get_dimension_by_id(ht->correlated_space, slice->fd.dimension_id);
+		if (dim == NULL)
+			elog(ERROR, "correlated constraint dimension %d not found!", slice->fd.dimension_id);
+
+		attno = ts_map_attno(ht->main_table_relid, chunk->table_id, dim->column_attno);
+
+		if (reset)
+		{
+			minmax[0] = Int64GetDatum(DIMENSION_SLICE_MINVALUE);
+			minmax[1] = Int64GetDatum(DIMENSION_SLICE_MAXVALUE);
+		}
+
+		/* calculate the min/max for this correlated constraint attribute on this chunk */
+		if (reset || ts_chunk_get_minmax(chunk->table_id,
+										 dim->fd.column_type,
+										 attno,
+										 "correlated constraint",
+										 minmax))
+		{
+			DimensionSlice *new_slice;
+			DimensionVec *vec;
+			int64 min = DatumGetInt64(minmax[0]);
+			int64 max = DatumGetInt64(minmax[1]);
+
+			/* The end value is exclusive to the range, so incr by 1 */
+			if (max != DIMENSION_SLICE_MAXVALUE)
+			{
+				max++;
+				/* Again, check overflow */
+				max = REMAP_LAST_COORDINATE(max);
+			}
+
+			/*
+			 * It's possible that an existing chunk might have the SAME values for this
+			 * correlated constraint due to which an entry might already exist in the
+			 * dimension_slice catalog table. So check for that.
+			 */
+
+			vec = ts_dimension_slice_equal_scan(dim->fd.id, min, max);
+
+			/* Add a new entry if none exists */
+			if (vec->num_slices == 0)
+			{
+				new_slice = ts_dimension_slice_create(dim->fd.id, min, max);
+				dimension_slice_insert(new_slice);
+			}
+			else
+			{
+				Assert(vec->num_slices == 1);
+				new_slice = vec->slices[0];
+			}
+
+			/* Generate a new name for this constraint using the new slice id */
+			snprintf(NameStr(constraint->fd.constraint_name),
+					 NAMEDATALEN,
+					 "%sconstraint_%d",
+					 CC_DIM_PREFIX,
+					 new_slice->fd.id);
```

Comment:
Why don't using the existing `chunk_constraint_dimension_choose_name` function for it?

Reply:
Yeah, I asked a similar question about the new prefix we use. Not saying its wrong, just wondering why we are not hooking into existing code or postgres  functions for naming.